### PR TITLE
[plugin-menu] internal: e2e test for plugin menu

### PR DESCRIPTION
**Pull request by danarish4**

### CODE REVIEW

File: examples/main/shared/editor/Editor.jsx [hunk 1]
```diff
@@ -30,18 +30,33 @@ export default class Editor extends PureComponent {
...
       ? testAppPlugins.map(plugin => Plugins.editorPluginsMap[plugin]).flat()
       : Plugins.editorPlugins;
     this.config = pluginsConfig;
-    this.toolbarsConfig = toolbarsConfig;
+    this.toolbarsConfig = this.getToolbarConfig();
+  }
+
+  getToolbarConfig() {
+    const { testApptoolbarConfig } = this.props;
+    console.log('4343434434', this.props);
```
Comment (shaulgo) [inline examples/main/shared/editor/Editor.jsx:43]: remove this

File: e2e/cypress/support/commands.js [hunk 2]
```diff
@@ -48,6 +49,27 @@ const run = (app, fixtureName, plugins) => {
 let isMobile = false;
 let isHebrew = false;
 let isSeoMode = false;
+let testApptoolbarConfig = [];
+
+Cypress.Commands.add('setHorizontalPluginMenu', () => {
```
Comment (shaulgo) [inline e2e/cypress/support/commands.js:54]: this doesn't seem like it is very convenient for all configs. We need to figure out a way that will be scaleable across all configs. Let's discuss tomorrow.

File: examples/main/shared/editor/Editor.jsx [hunk 3]
```diff
@@ -30,18 +30,33 @@ export default class Editor extends PureComponent {
...
+    if (testApptoolbarConfig) {
```
Comment (shaulgo) [inline examples/main/shared/editor/Editor.jsx:44]: this should get the config as an object.

File: e2e/tests/pluginMenu.e2e.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      cy.eyesCheckWindow(this.test.title);
+    });
+    it('should render advanced plugin menu with sections', function() {
+      cy.setPluginMenuSectionSplited();
+      cy.loadEditorAndViewer('newLines')
+        .focusEditor()
+        .openSideToolbar();
+      cy.eyesCheckWindow(this.test.title);
+    });
+    it('should render advanced plugin menu with sections and search', function() {
```
Comment (danarish4) [inline e2e/tests/pluginMenu.e2e.js:45]: plugin menu & horizontal plugin

File: examples/main/shared/editor/Editor.jsx [hunk 5]
```diff
@@ -28,20 +28,34 @@ export default class Editor extends PureComponent {
...
       : Plugins.editorPlugins;
     this.config = pluginsConfig;
-    this.toolbarsConfig = toolbarsConfig;
+    this.toolbarsConfig = this.getToolbarConfig();
+  }
+
+  getToolbarConfig() {
+    const { testAppConfig = {} } = this.props;
+    if (testAppConfig.toolbarConfig) {
+      if (testAppConfig.toolbarConfig.includes('horizontal')) {
```
Comment (danarish4) [inline examples/main/shared/editor/Editor.jsx:44]: fix

Comment (danarish4) [reply examples/main/shared/editor/Editor.jsx:44]: toolbarConfig = undefined

File: e2e/tests/plugins.e2e.js [hunk 6]
```diff
@@ -219,7 +224,7 @@ describe('plugins', () => {
 
     beforeEach('load editor', () => {
       cy.switchToDesktop();
-      cy.loadEditorAndViewer('empty', 'linkPreview');
+      cy.loadEditorAndViewer('empty', getPluginsToConsumeConfig('linkPreview'));
```
Comment (danarish4) [inline e2e/tests/plugins.e2e.js:227]: useplugins

Comment (danarish4) [reply e2e/tests/plugins.e2e.js:227]: change to const insttead 'linkPreview'

File: examples/main/shared/editor/Editor.jsx [hunk 7]
```diff
@@ -28,20 +28,28 @@ export default class Editor extends PureComponent {
...
+    this.plugins = testAppConfig.plugins
+      ? testAppConfig.plugins.map(plugin => Plugins.editorPluginsMap[plugin]).flat()
       : Plugins.editorPlugins;
     this.config = pluginsConfig;
-    this.toolbarsConfig = toolbarsConfig;
+    this.toolbarsConfig = this.getToolbarConfig();
+  }
+
+  getToolbarConfig() {
+    const { testAppConfig = {} } = this.props;
```
Comment (danarish4) [inline examples/main/shared/editor/Editor.jsx:42]: delete function

**Pull request merged**